lib/msf/base/simple/exception.rb

```diff
@@ -0,0 +1,28 @@
+# -*- coding: binary -*-
+
+module Msf
+  module Simple
+    module Exception
```

Comment:
Then namespace here suggests this class is used as an `Exception` when it is in just a formatter.  This is currently only used with `::Msf::OptionValidateError` would it make sense to simply add the method in `::Msf::OptionValidateError`?

Reply:
I would even hazard the signature could change to allow usage as:
```
rescue ::Msf::OptionValidateError => e
  mod.print_error(e.format_validation_error)
```

This avoids the need to pass in the `mod` object and allows the method to access `self.options` and `self.reasons`.

Reply:
I originally had a string being returned, but decided to take the entire mod + error, to allow the flexibility of calling `print_error` / `print_status` etc, as I think this implementation will change over time.

Reply:
I would prefer to see this changed before this lands. From a code responsibilities & naming perspective, the class name inferring it is an Exception and need to inject the module are non-starters IMHO.